Assets/MixedRealityToolkit/_Core/Definitions/Devices/InteractionMapping.cs

```diff
@@ -244,4 +243,129 @@ public void SetValue(Tuple<Vector3, Quaternion> newValue)
 
         #endregion Set Operators
     }
+
+    public class Test
+    {
+        System.Collections.Generic.List<IInteractionMapping> list = new System.Collections.Generic.List<IInteractionMapping>();
+        void helloWorld()
+        {
+            var interaction = new InteractionMapping<float>(0, AxisType.SingleAxis, DeviceInputType.None, new InputAction(1, "None"));
+            var value = interaction.GetData();
+            interaction.SetData(0f);
+            list.Add(interaction);
+            InteractionMapping<float> interaction2 = (InteractionMapping<float>)list[0];
+            var value2 = interaction2.GetData();
+            interaction2.SetData(0.1f);
+        }
+    }
+
+    public interface IInteractionMapping
+    {
+        /// <summary>
+        /// The Id assigned to the Interaction.
+        /// </summary>
+        uint Id { get; }
+
+        /// <summary>
+        /// The axis type of the button, e.g. Analogue, Digital, etc.
+        /// </summary>
+        AxisType AxisType { get; }
+
+        /// <summary>
+        /// The primary action of the input as defined by the controller SDK.
+        /// </summary>
+        DeviceInputType InputType { get; }
+
+        /// <summary>
+        /// Action to be raised to the Input Manager when the input data has changed.
+        /// </summary>
+        InputAction InputAction { get; }
+
+        /// <summary>
+        /// Has the value changed since the last reading.
+        /// </summary>
+        bool Changed { get; }
+    }
+
+    public class InteractionMapping<TReadingType> : IInteractionMapping
+    {
+        public InteractionMapping(uint id, AxisType axisType, DeviceInputType inputType, InputAction inputAction)
+        {
+            this.id = id;
```

Comment:
would be great to add a check that TReadingType is compatible with the specified AxisType. if we are certain that this is internal MRTK _Core code only, then an Assert will do.

Reply:
I'll have a look, however Checking the type at runtime will incur a boxing cost most likely.
Main point is it shouldn't matter using this implementation. Main reason for the check in the old version was that we had multiple datum storage, now we only have one. 

The type set when the mapping was created enforces the data type stored.  Although I suppose we could have a "Validator" instead of building it in to the mapping.